telemetry/packages/public-app/components/levitationHeight.tsx

```diff
@@ -0,0 +1,206 @@
+        Object.keys(data)
+          .slice(0, Object.keys(data).length - 2)
+          .map(height => (
+            <div key={height} className="space-y-2 mt-4">
+              <Flex>
+                <Text>{height}</Text>
+                <Text>{`${
+                  data[height][0]?.value ? data[height][0].value : 0
+                } mm`}</Text>
+              </Flex>
+
+              <ProgressBar value={data[height][0]?.value}></ProgressBar>
+            </div>
+          ))
+      ) : (
+        <div>no data </div>
+      )}
+
+      {show ? (
+        <div>
+          {Object.keys(data)
+            .slice(-2)
+            .map(height => (
+              <div key={height} className="space-y-2 mt-4">
+                <Flex>
+                  <Text>{height}</Text>
+                  <Text>{`${
+                    data[height][0]?.value ? data[height][0].value : 0
+                  } mm`}</Text>
+                </Flex>
+
+                <ProgressBar value={data[height][0]?.value}></ProgressBar>
+              </div>
+            ))}
+        </div>
+      ) : null}
+      <Flex className="mt-6 pt-4 border-t">
+        <Button
+          size="xs"
+          variant="light"
+          //   icon={ArrowNarrowRightIcon}
+          iconPosition="right"
+          onClick={() => {
+            !show ? setShow(true) : setShow(false);
+          }}
+        >
+          View more
+        </Button>
+      </Flex>
+     
+      {show ? null : !(Object.keys(data)[0] === 'statusCode') ? (
+        (console.log('no error', error),
+        console.log('This is data statuscode ' + typeof data['statusCode']),
+        (
+          <div className="grid grid-cols-2 gap-2 p-5 ml-[-14px] levitation-badges">
+            {Object.keys(data).map((height, i) => (
+              <BadgeDelta
+                key={i}
+                deltaType={
+                  data[height][0]?.value > 0
+                    ? 'moderateIncrease'
+                    : 'moderateDecrease'
+                }
+              >
+                Sensor {i}:{' '}
+                {data[height][0]?.value > 0 ? 'Elevated' : 'disconnected'}
+              </BadgeDelta>
+            ))}
+          </div>
+        ))
+      ) : (
+        <div className="grid grid-cols-2 gap-2 p-5 ml-[-14px] levitation-badges">
+          <BadgeDelta key={1} deltaType="moderateDecrease">
+            Disconnected
+          </BadgeDelta>
+          <BadgeDelta key={2} deltaType="moderateDecrease">
+            Disconnected
+          </BadgeDelta>
+          <BadgeDelta key={3} deltaType="moderateDecrease">
+            Disconnected
+          </BadgeDelta>
+          <BadgeDelta key={4} deltaType="moderateDecrease">
+            Disconnected
+          </BadgeDelta>
+          <BadgeDelta key={5} deltaType="moderateDecrease">
+            Disconnected
+          </BadgeDelta>
+          <BadgeDelta key={6} deltaType="moderateDecrease">
+            Disconnected
+          </BadgeDelta>
```

Comment:
Could simplify this by having a loop for 6 items. Would also be good to have this dynamic - what if we have 10 levitation height sensors in the future?

Reply:
true

Reply:
Any update on this?